### PR TITLE
LPS-194431 When managing a folder of objects via Model Builder when we try to Edit in page view it is still possible to interact with the objects in the diagram

### PR DESCRIPTION
**Pull request by Artartur**
Related Issue: [LPS-194431](https://liferay.atlassian.net/browse/LPS-194431)

### CODE REVIEW

**Comment by Artartur** [comment]
ci:test:sf

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ObjectDefinitionNode/ObjectDefinitionNode.tsx [hunk 1]
```diff
@@ -144,15 +144,38 @@ export function ObjectDefinitionNode({
...
+		dispatch({
+			payload: {
+				draggable: false,
+				objectDefinitionNodes: nodes,
+				objectRelationshipEdges: edges,
+				selectedObjectDefinitionId: id.toString(),
+			},
+			type: TYPES.SET_SELECTED_OBJECT_DEFINITION_NODE_DRAGGRABLE,
+		});
 	};
```
Comment (igor-franca) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ObjectDefinitionNode/ObjectDefinitionNode.tsx:179]: Hey @Artartur, could you reorder this function and add a blank space after `const` declaration? i.e

```
const handleShowRedirectObjectDefinitionModal = () => {
  const {edges, nodes} = store.getState();

  dispatch({
    payload: {
      draggable: false,
      objectDefinitionNodes: nodes,
      objectRelationshipEdges: edges,
      selectedObjectDefinitionId: id.toString(),
    },
    type: TYPES.SET_SELECTED_OBJECT_DEFINITION_NODE_DRAGGRABLE,
  });

  setShowModal({
      redirectToEditObjectDefinitionDetails: true,
    });
  };
```

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ObjectDefinitionNode/ObjectDefinitionNode.tsx [hunk 2]
```diff
@@ -144,15 +144,38 @@ export function ObjectDefinitionNode({
...
+
+		dispatch({
+			payload: {
+				draggable: false,
+				objectDefinitionNodes: nodes,
+				objectRelationshipEdges: edges,
+				selectedObjectDefinitionId: id.toString(),
+			},
+			type: TYPES.SET_SELECTED_OBJECT_DEFINITION_NODE_DRAGGRABLE,
+		});
```
Comment (igor-franca) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ObjectDefinitionNode/ObjectDefinitionNode.tsx:161]: Do the same here

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ObjectDefinitionNode/ObjectDefinitionNode.tsx [hunk 3]
```diff
@@ -401,12 +424,24 @@ export function ObjectDefinitionNode({
...
+						dispatch({
+							payload: {
+								draggable: true,
+								objectDefinitionNodes: nodes,
+								objectRelationshipEdges: edges,
+								selectedObjectDefinitionId: id.toString(),
+							},
+							type:
+								TYPES.SET_SELECTED_OBJECT_DEFINITION_NODE_DRAGGRABLE,
+						});
```
Comment (igor-franca) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ObjectDefinitionNode/ObjectDefinitionNode.tsx:444]: Do the same here

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ObjectDefinitionNode/ObjectDefinitionNode.tsx [hunk 4]
```diff
@@ -448,9 +483,21 @@ export function ObjectDefinitionNode({
...
```
Comment (igor-franca) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ObjectDefinitionNode/ObjectDefinitionNode.tsx:500]: Do the same here

File: modules/apps/object/object-web/types/src/main/resources/META-INF/resources/js/components/ModelBuilder/types.d.ts [hunk 5]
```diff
@@ -117,6 +117,15 @@ export declare type TAction =
 			};
 			type: TYPES.SET_SELECTED_OBJECT_DEFINITION_NODE;
 	  }
+	| {
+			payload: {
+				draggable: boolean;
```
Comment (igor-franca) [inline modules/apps/object/object-web/types/src/main/resources/META-INF/resources/js/components/ModelBuilder/types.d.ts:122]: Hey @Artartur, you could rename this propertie to `objectDefinitionNodeDraggable` to be consistency with `SET_SELECTED_OBJECT_DEFINITION_NODE_DRAGGRABLE`

**Review by igor-franca — CHANGES_REQUESTED** [review]
Hey @Artartur, good work with this PR! Just some comments

**Comment by Artartur** [comment]
ci:test:sf

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ObjectDefinitionNode/ObjectDefinitionNode.tsx (hunk 1 above)
Comment (Artartur) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ObjectDefinitionNode/ObjectDefinitionNode.tsx]: done!

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ObjectDefinitionNode/ObjectDefinitionNode.tsx (hunk 2 above)
Comment (Artartur) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ObjectDefinitionNode/ObjectDefinitionNode.tsx:157]: done!

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ObjectDefinitionNode/ObjectDefinitionNode.tsx (hunk 3 above)
Comment (Artartur) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ObjectDefinitionNode/ObjectDefinitionNode.tsx:439]: done!

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ObjectDefinitionNode/ObjectDefinitionNode.tsx (hunk 4 above)
Comment (Artartur) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ObjectDefinitionNode/ObjectDefinitionNode.tsx:499]: done!

File: modules/apps/object/object-web/types/src/main/resources/META-INF/resources/js/components/ModelBuilder/types.d.ts (hunk 5 above)
Comment (Artartur) [reply modules/apps/object/object-web/types/src/main/resources/META-INF/resources/js/components/ModelBuilder/types.d.ts]: done!

**Comment by igor-franca** [comment]
hey @Artartur, for this fix to be more complete, could you cover delete, relationship and field modals with this fix?

**Comment by Artartur** [comment]
ci:test:sf

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ObjectDefinitionNode/ObjectDefinitionNodeFooter.tsx [hunk 6]
```diff
@@ -22,10 +26,16 @@ interface ObjectDefinitionNodeFooterProps {
...
 	handleSelectObjectDefinitionNode,
 	isLinkedObjectDefinition,
+	selectedObjectDefinitionId,
 	setShowAllObjectFields,
 	setShowModal,
 	showAllObjectFields,
 }: ObjectDefinitionNodeFooterProps) {
+	const [_, dispatch] = useObjectFolderContext();
+
+	const store = useStore();
```
Comment (igor-franca) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ObjectDefinitionNode/ObjectDefinitionNodeFooter.tsx:36]: Hey @Artartur, when we create a constant that will be used below, we usually add a white line between the definition and the use, so:

```
	const [_, dispatch] = useObjectFolderContext();
	const store = useStore();
	
	const {edges, nodes} = store.getState();
```
Please find other occurrences and do the same

**Review by igor-franca — CHANGES_REQUESTED** [review]
Very good @Artartur, we're almost there!

**Comment by Artartur** [comment]
ci:test:sf

**Comment by Artartur** [comment]
ci:test:sf

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ModelBuilderContext/objectFolderContext.tsx [hunk 7]
```diff
@@ -29,6 +29,20 @@ interface ObjectFolderContextProviderProps
...
+		addObjectRelationship: false,
+		deleteObjectDefinition: false,
+		deleteObjectFolder: false,
+		deleteObjectRelationship: false,
+		editObjectDefinitionExternalReferenceCode: false,
+		editObjectFolder: false,
+		moveObjectDefinition: false,
+		publishObjectDefinitions: false,
+		redirectToEditObjectDefinitionDetails: false,
+	},
```
Comment (igor-franca) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/ModelBuilderContext/objectFolderContext.tsx:45]: Hey @Artartur, could you please rename `changeModalVisibility` to `modelBuilderModals` and use the interface `ModelBuilderModals` instead declare all the properties? So:

`modelBuilderModals: {} as ModelBuilderModals`

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx [hunk 8]
```diff
@@ -33,34 +53,101 @@ export default function EditObjectFolder({
...
 		dispatch,
 	] = useObjectFolderContext();
 
+	const selectedObjectDefinitionNodeId =
+		selectedObjectDefinitionNode?.data?.id ?? 0;
+
+	const viewObjectDetailsURL = formatActionURL(
+		editObjectDefinitionURL,
+		selectedObjectDefinitionNodeId
+	);
```
Comment (igor-franca) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx:77]: We can use formatActionURL inline, so:

```
						viewObjectDetailsURL={formatActionURL(
							editObjectDefinitionURL,
							selectedObjectDefinitionNode?.data?.id
						)}
```

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx [hunk 9]
```diff
@@ -33,34 +53,101 @@ export default function EditObjectFolder({
...
 			rightSidebarType,
+			selectedObjectDefinitionNode,
 			selectedObjectFolder,
 			showChangesSaved,
 		},
 		dispatch,
 	] = useObjectFolderContext();
 
+	const selectedObjectDefinitionNodeId =
+		selectedObjectDefinitionNode?.data?.id ?? 0;
```
Comment (igor-franca) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx:72]: Hey @Artartur, I believe this `const` is not necessary after new changes.

**Review by igor-franca — CHANGES_REQUESTED** [review]
Good job with this PR, just some improvements!

**Comment by Artartur** [comment]
ci:test:sf

**Comment by Artartur** [comment]
ci:test:sf

**Comment by Artartur** [comment]
ci:test:sf

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx [hunk 10]
```diff
@@ -33,34 +53,94 @@ export default function EditObjectFolder({
...
-		deleteObjectDefinition: false,
-		deleteObjectFolder: false,
-		deleteObjectRelationship: false,
-		editObjectDefinitionExternalReferenceCode: false,
-		editObjectFolder: false,
-		moveObjectDefinition: false,
-		publishObjectDefinitions: false,
-		redirectToEditObjectDefinitionDetails: false,
-	});
+	const {edges, nodes} = store.getState();
```
Comment (igor-franca) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx:86]: Declared here

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx [hunk 11]
```diff
@@ -137,34 +222,227 @@ export default function EditObjectFolder({
...
+							?.defaultLanguageId as Liferay.Language.Locale
+					}
+					objectDefinitionExternalReferenceCode={
+						selectedObjectDefinitionNodeERC
+					}
+					objectDefinitionName={
+						selectedObjectDefinitionNode?.data?.name as string
+					}
+					onAfterSubmit={(newObjectField) => {
+						const {edges, nodes} = store.getState();
```
Comment (igor-franca) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx:239]: Hey @Artartur, the `edges` and `nodes` has already be created in global scope of this file. So you can remove this repeated declaration here.

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx [hunk 12]
```diff
@@ -33,34 +53,94 @@ export default function EditObjectFolder({
...
 		dispatch,
 	] = useObjectFolderContext();
 
+	const [
+		objectRelationshipParameterRequired,
+		setObjectRelationshipParameterRequired,
+	] = useState(false);
+
+	const selectedObjectDefinitionNodeERC = selectedObjectDefinitionNode?.data
+		?.externalReferenceCode as string;
```
Comment (igor-franca) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx:77]: Hey @Artartur, can you use the `selectedObjectDefinitionNode?.data?.externalReferenceCode` directly

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx [hunk 13]
```diff
@@ -137,34 +222,227 @@ export default function EditObjectFolder({
 				/>
 			)}
 
-			{showModal.editObjectFolder && (
+			{modelBuilderModals.addObjectField && (
+				<ModalAddObjectField
+					baseResourceURL={baseResourceURL}
+					creationLanguageId={
+						selectedObjectDefinitionNode?.data
+							?.defaultLanguageId as Liferay.Language.Locale
```
Comment (igor-franca) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx:230]: Hey @Artartur, can you use `selectedObjectDefinitionNode.data?`  as a modal rendering condition. So you don't need use `as Liferay.Language.Locale`, `as string` and others.

Apply the same patten to athers modals

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx [hunk 14]
```diff
@@ -137,34 +222,227 @@ export default function EditObjectFolder({
...
+					objectDefinitionExternalReferenceCode={
+						selectedObjectDefinitionNodeERC
+					}
+					objectDefinitionName={
+						selectedObjectDefinitionNode?.data?.name as string
+					}
+					onAfterSubmit={(newObjectField) => {
+						const {edges, nodes} = store.getState();
+
+						if (selectedObjectDefinitionNode) {
```
Comment (igor-franca) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx:241]: > Hey @Artartur, can you use selectedObjectDefinitionNode.data? as a modal rendering condition. So you don't need use as Liferay.Language.Locale, as string and others.

You can remove this `if` after the change above

**Review by igor-franca — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx (hunk 14 above)
Comment (Artartur) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx]: Done!

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx (hunk 13 above)
Comment (Artartur) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx]: Done!

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx (hunk 12 above)
Comment (Artartur) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx]: Done!

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx (hunk 11 above)
Comment (Artartur) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx]: Done!

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx (hunk 10 above)
Comment (Artartur) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx:78]: Done!

**Comment by Artartur** [comment]
ci:test:sf

**Review by rebsilva — APPROVED** [review]
LGTM! :tada:

**Comment by rebsilva** [comment]
ci:test:relevant

**Comment by Artartur** [comment]
ci:test:sf

**Comment by Artartur** [comment]
ci:test:objects

**Comment by Artartur** [comment]
ci:test:object

**Comment by Artartur** [comment]
ci:test:sf

**Comment by Artartur** [comment]
ci:forward

**Comment by Artartur** [comment]
ci:foward:force

**Comment by diogomelo09** [comment]
ci:foward:force

**Comment by Artartur** [comment]
ci:forward:force

**Pull request closed without merging**